docs/Integrating Your HMI/Vehicle Data/index.md

```diff
@@ -0,0 +1,169 @@
+                    "name": "customStructVal",
+                    "key": "KEY_CUSTOM_STRUCT_VAL",
+                    "type": "String",
+                    "mandatory": true
+                },
+                {
+                    "name": "customStructVal2",
+                    "key": "KEY_CUSTOM_STRUCT_VAL2",
+                    "minvalue": 0,
+                    "maxvalue": 100,
+                    "type": "Integer",
+                    "mandatory": true
+                },
+                {
+                    "name": "customDeprecatedVal",
+                    "key": "KEY_CUSTOM_DEPRECATED_VAL",
+                    "minvalue": 0,
+                    "maxvalue": 100,
+                    "type": "Integer",
+                    "mandatory": true,
+                    "until": "7.0"
+                },
+                {
+                    "name": "customDeprecatedVal",
+                    "key": "KEY_CUSTOM_DEPRECATED_VAL",
+                    "minvalue": 0,
+                    "maxvalue": 100,
+                    "type": "Integer",
+                    "mandatory": true,
+                    "deprecated": true
+                    "since": "7.0"
+                }
+            ],
+            "key": "KEY_CUSTOM_STRUCT",
+            "type": "Struct",
+            "mandatory": false
+        }
+    ]
+}
+```
+
+In addition to custom items, this feature can be used to expose other vehicle data items that were introduced to the project in later versions. This can be useful when the software version on the head unit cannot be updated easily. If a vehicle data item is added into the project, the definition of this item will be included in the policy table by default. Any vehicle data items which are defined in Core's local Mobile API will be ignored from the policy table, but newer items will be interpreted as custom items. This allows apps to use these data items normally if they are exposed by the head unit, even when they were not initially supported.
+
+### Custom Vehicle Data Requests
+
+Custom vehicle data requests have a separate structure to normal vehicle data requests. While normal vehicle data items are requested using the key structure of `"<item.name>: true"`, custom items are constructed using the `item.key` field and can have a nested structure (when requesting `Struct` items). For example, when requesting all of the vehicle data items which are defined above, the HMI would receive the following message:
+
+```
+{
+    "id" : 139,
+    "jsonrpc" : "2.0",
+    "method" : "VehicleInfo.GetVehicleData",
+    "params" : {
+        "KEY_CUSTOM_STRING": true,
+        "KEY_CUSTOM_INT": true,
+        "KEY_CUSTOM_FLOAT": true,
+        "KEY_CUSTOM_BOOL": true,
+        "KEY_CUSTOM_ARRAY": true,
+        "KEY_CUSTOM_STRUCT": {
+            "KEY_CUSTOM_STRUCT_VAL": true,
+            "KEY_CUSTOM_STRUCT_VAL2": true,
+            "KEY_CUSTOM_DEPRECATED_VAL": true
+        }
+    }
+}
+```
+
+### Vehicle Data Mapping File
+
```

Comment:
should be mentioned that GetPolicyConfigurationData is needed to get the OEM mapping table version

Reply:
- HMI sends GetPolicyConfigurationData request to core to read OEM Network Mapping version from module_config -> endpoint_properties
- Core responds with available endpoint_properties
- HMI parses the version for custom_vehicle_data_mapping_url and compares that with version of local copy of OEM Network Mapping table file and decides if OEM Network Mapping table needs to be updated
- HMI sends GetPolicyConfigurationData request to core to read url from endpoint object for OEM Network Mapping table
- Core responds with endpoints and HMI parses the url for service custom_vehicle_data_mapping_url
- HMI sends onSystemRequest with requestType OEM_SPECIFIC and requestSubType VEHICLE_DATA_MAPPING to core which core forwards to Mobile app (via proxy)
- Mobile app downloads the OEM Network Mapping table and sends to core as bulk data in SystemRequest requestType OEM_SPECIFIC and requestSubType VEHICLE_DATA_MAPPING which core forwards to HMI

Reply:
It's included in a note below:
```
!!! NOTE
In order for the HMI to determine when this file needs to be updated, this file can be assigned a version via the `module_config.endpoint_properties.custom_vehicle_data_mapping.version` field. The HMI can retrieve this field using the [SDL.GetPolicyConfigurationData](https://smartdevicelink.com/en/guides/hmi/sdl/getpolicyconfigurationdata/) RPC.
!!!
```